liblustreapi/src/lib.rs

```diff
@@ -305,7 +305,19 @@ impl Llapi {
     }
 
     pub fn search_fsname(&self, pathname: &str) -> Result<String, LiblustreError> {
-        let fsc = CString::new(pathname.as_bytes())?;
+        let mut path = PathBuf::from(pathname);
+        let md = std::fs::symlink_metadata(&path)?;
+        if md.file_type().is_symlink() {
+            path.pop();
+        }
+
+        let fsc = CString::new(
+            path.clone()
+                .into_os_string()
+                .into_string()
+                .unwrap()
+                .as_bytes(),
+        )?;
```

Comment:
```suggestion
        let fsc = CString::new(path.into_os_string().as_bytes())?;
```

(Import `std::os::unix::ffi::OsStrExt` at the top of the file)

Reply:
Done